packages/components/organisms/f-checkout/src/components/_tests/Checkout.test.js

```diff
@@ -3210,6 +3237,142 @@ describe('Checkout', () => {
                 });
             });
         });
+
+        describe('getReferralState ::', () => {
+            afterEach(() => {
+                jest.clearAllMocks();
+            });
+
+            it('should retrieve the referral state from the cookie', () => {
+                // Arrange
+                const getSpy = jest.fn().mockImplementation(() => ({ menuReferralState: 1 }));
+
+                const wrapper = shallowMount(VueCheckout, {
+                    store: createStore({ ...defaultCheckoutState, restaurant: { id: '99999' } }),
+                    i18n,
+                    localVue,
+                    propsData,
+                    mocks: {
+                        $logger,
+                        $cookies: {
+                            get: getSpy
```

Comment:
I dont think you need to point the internal `get` to the spy here? You should be able to just create the spy above from the wrapper instance and then use that the assert. The `get` will already be tied to the `jest.fn()` via the setup file anyway so here you can just have `$cookies` i believe.
```javascript
jest.spyOn(wrapper.vm.$cookies, 'get').mockReturnValue(() => { menuReferralState: 1 });

mocks: {
  $cookies
}

expect(getSpy).toHaveBeenCalledWith('je-rw-menu-referral-state-99999');
```

Reply:
Nope, I just tried this and the test failed. I don't see why it would work, though. But maybe I'm missing something?

Reply:
What was the failure out of interest? I did the same sort of thing earlier [here](https://github.com/justeat/fozzie-components/pull/1034/files#diff-77d6d5e41e52cc7565e75f71e027e67a362fae9f1f6e0c8954c6b95ea57f8d50R124)

Reply:
It should work because you're passing the $cookie instance as a mock to the wrapper so when you spy on it you're spying on the jest instance. Happy to approve it for some reason it's not working for you.

Reply:
Ahhh, wait, I understand what you did now. Ok, I'll implement this 👍